include/deal.II/particles/particle_handler.h

```diff
@@ -619,6 +750,58 @@ namespace Particles
       &global_number_of_particles &global_max_particles_per_cell
         &                          next_free_particle_index;
   }
+
+
+
+  template <int dim, int spacedim>
+  template <class VectorType>
+  void
+  ParticleHandler<dim, spacedim>::set_particle_positions(
+    const VectorType &input_vector,
+    const bool        displace_particles)
+  {
+    AssertDimension(input_vector.size(),
+                    get_next_free_particle_index() * spacedim);
+    for (auto &p : *this)
+      {
+        const auto &point     = p.get_location();
+        auto        new_point = point * (displace_particles ? 1.0 : 0.0);
+        const auto  id        = p.get_id();
+        for (unsigned int i = 0; i < spacedim; ++i)
+          new_point[i] += input_vector[id * spacedim + i];
+        p.set_location(new_point);
+      }
+    sort_particles_into_subdomains_and_cells();
+  }
+
+
+
+  template <int dim, int spacedim>
+  template <class VectorType>
+  void
+  ParticleHandler<dim, spacedim>::get_particle_positions(
+    VectorType &output_vector,
+    const bool  add_to_output_vector)
+  {
+    AssertDimension(output_vector.size(),
+                    get_next_free_particle_index() * spacedim);
+    for (const auto &p : *this)
+      {
+        auto &     point = p.get_location();
+        const auto id    = p.get_id();
+        if (add_to_output_vector)
+          for (unsigned int i = 0; i < spacedim; ++i)
+            output_vector[id * spacedim + i] += point[i];
+        else
+          for (unsigned int i = 0; i < spacedim; ++i)
+            output_vector[id * spacedim + i] = point[i];
+      }
+    if (add_to_output_vector)
+      output_vector.compress(VectorOperation::add);
+    else
+      output_vector.compress(VectorOperation::insert);
```

Comment:
Ah, so it's a very specific kind of vector type that supports the `compress()` operation.

Reply:
All of our vector types support the compress() operation.